packages/core/src/lib/markdown-it/highlight/HighlightRule.js

```diff
@@ -0,0 +1,82 @@
+const { HighlightRuleComponent } = require('./HighlightRuleComponent.js');
+
+class HighlightRule {
+  constructor(rule) {
+    /**
+     * @type Array<HighlightRuleComponent>
+     */
+    this.rule = rule;
+  }
+
+  isLineNumber() {
+    return this.rule.length === 1 && this.rule[0].isNumber()
+  }
+  
+  isLineSlice() {
+    return this.rule.length === 1 && this.rule[0].isSlice();
```

Comment:
Since its used only in `applyHighlight` and is rather simple, perhaps we could just use a named constant instead? (ignore if this is going to be removed when dealing with the below)